src/context.js

```diff
@@ -1,10 +1,17 @@
 "use strict";
 module.exports = function(Promise) {
+
+var util = require("./util");
 var longStackTraces = false;
 var contextStack = [];
 
-Promise.prototype._promiseCreated = function() {};
-Promise.prototype._pushContext = function() {};
+util.hookTo(Promise.prototype, "_hook_created", null);
```

Comment:
Given all these changes - can you please run the benchkmarks before/after this PR and with montioring turned on and off and report the results? 

We need to be extra careful that users with monitoring turned off do not have to pay for it.